### PR TITLE
SPLICE-1763 Fix select from partitioned orc table error

### PR DESCRIPTION
**Pull request by tgildersleeve**
previously you couldn't select from an orc table that was partitioned by any column other than the last column.

### CODE REVIEW

File: db-engine/src/main/java/com/splicemachine/db/iapi/sql/execute/ExecRow.java [hunk 1]
```diff
@@ -111,6 +111,8 @@
 
 	StructType createStructType();
```
Comment (jleach4) [inline db-engine/src/main/java/com/splicemachine/db/iapi/sql/execute/ExecRow.java]: Why is this required vs. createStructType?  Not clear from comments or code.

Comment (tgildersleeve) [inline db-engine/src/main/java/com/splicemachine/db/iapi/sql/execute/ExecRow.java]: Depending on which columns you select from an orc table, createStructType can return the incorrect columns because it only has access to the total number N of columns selected, and returns the first N columns in the table rather than the correct N. Passing in ColumnBaseMap to createStructTypeCorrected allows the correct N columns to be returned.
